packages/ember-data/lib/system/model/internal-model.js

```diff
@@ -663,6 +663,42 @@ InternalModel.prototype = {
     this._inFlightAttributes = Ember.create(null);
   },
 
+  /**
+    @method _changedKeys
+    @private
+  */
+  _changedKeys: function(updates) {
```

Comment:
Add a comment here to describe this algorithm because it can be tricky to understand when revisiting.